beekeeper-core/src/main/java/com/expediagroup/beekeeper/core/repository/HousekeepingPathRepository.java

```diff
@@ -34,4 +37,14 @@
       + "and p.modifiedTimestamp <= :instant order by p.modifiedTimestamp")
   Page<EntityHousekeepingPath> findRecordsForCleanupByModifiedTimestamp(@Param("instant") LocalDateTime instant,
       Pageable pageable);
+
+  @Modifying
+  @Transactional
+  @Query("delete from EntityHousekeepingPath p where "
+      + " p.databaseName=:databaseName and "
+      + " p.tableName=:tableName and "
+      + " p.lifecycleType = 'EXPIRED'")
+  void cleanupOldExpiredRows(
```

Comment:
As per the other comment, this should be an update operation instead

Reply:
I replaced this with an incode upsert given that we'd most likely only want to update the
path, cleanupTimestamp or cleanupDelay fields. I had some sql written out for the upsert, but it seemed a bit convoluted and harder to maintain longterm than wrapping the functionality within a method.

It's at `PathSchedulerService#scheduleExpiration` now.